Effisense/sensor-dashboard
# Add user and organization to database on sign in and when setting active organization

Closes: #29, #30, #40 

## Important to note

The `/customer` page is not entirely done before Effisense responds to [the Slack thread](https://effisensexjrc.slack.com/archives/C04ST2VCNHW/p1680185211016029). However, much of the functionality, route guarding and logic is in place. Hence, I think this is subject to be merged when the review is OK, and then we'll create a separate issue on the actually content of `/customer`, and how the user should be added to an initial organization.

## Changelog

- Add user to MySQL database after login if they do not already exist
- Check if user is member of any organization, and display an organization switcher on `/customer` if so
- Add user to organization
- Set the active organization from `/customer`
- Handle `onSuccess` and `onError` in `/customer`
- Refactor all mutation to use `onSuccess` and `onError`, so to clean up our custom hooks

## How to test

- Ensure you are logged in, and unset your active organization. This means that you should be in your personal workspace.
- Log out
- Sign in
- Observe that, because you have no active organization, you are taken to `/customer`
- Set your active organization
- Observe that you are redirected to the dashboard (`/`)
- Open up Prisma Studio
- Inspect the `User`, `UserInOrganization` and `Organization` tables
- Observe that your user has been added, the organization has been added, and the link between them has also been added
